sweep/sweeper.go

```diff
@@ -1,40 +1,691 @@
+		}
+	}
+
+	// Create sets for just the new inputs.
+	newSets, err := generateInputPartitionings(
+		newInputs,
+		s.relayFeePerKW, satPerKW,
+		s.cfg.MaxInputsPerTx,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Append the new sets at the end of the list, because those tx likely
+	// have a higher fee per input.
+	return append(allSets, newSets...), nil
+}
+
+// sweep takes a set of preselected inputs, creates a sweep tx and publishes the
+// tx. The output address is only marked as used if the publish succeeds.
+func (s *UtxoSweeper) sweep(inputs []Input,
+	satPerKW lnwallet.SatPerKWeight, currentHeight int32) error {
+
+	var err error
+
+	// Generate output script if no unused script available.
+	if s.currentOutputScript == nil {
+		s.currentOutputScript, err = s.cfg.GenSweepScript()
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create sweep tx.
+	tx, err := createSweepTx(inputs,
+		s.currentOutputScript,
+		uint32(currentHeight), satPerKW, s.cfg.Signer,
+	)
+	if err != nil {
+		return fmt.Errorf("cannot create sweep tx: %v", err)
+	}
+
+	// Add tx before publication, so that we will always know that a spend
+	// by this tx is ours. Otherwise if the publish doesn't return, but did
+	// publish, we loose track of this tx. Even republication on startup
+	// doesn't prevent this, because that call returns a double spend error
+	// then and would also not add the hash to the store.
+	err = s.cfg.Store.NotifyPublishTx(tx)
+	if err != nil {
+		return fmt.Errorf("cannot notify publish: %v", err)
+	}
+
+	// Publish sweep tx.
+	log.Debugf("Publishing sweep tx %v", tx.TxHash())
+	err = s.cfg.PublishTransaction(tx)
+
+	// Keep outputScript in case of an error, so that it can be
+	// reused for the next tx and causes no address inflation.
+	if err != nil {
+		log.Error(err)
+	} else {
+		s.currentOutputScript = nil
+	}
+
+	// Reschedule sweep.
+	for _, input := range tx.TxIn {
+		pi, ok := s.pendingInputs[input.PreviousOutPoint]
+		if !ok {
+			// It can be that the input has been removed because it
+			// exceed the maximum number of attempts in a previous
+			// input set.
+			continue
+		}
+
+		// Record another publish attempt.
+		pi.publishAttempts++
+
+		// We don't care what the result of the publish call was. Even
+		// if it is published successfully, it can still be that it
+		// needs to be retried. Call NextAttemptDeltaFunc to calculate
+		// when to resweep this input.
+		nextAttemptDelta := s.cfg.NextAttemptDeltaFunc(
+			pi.publishAttempts)
+
+		pi.minPublishHeight = currentHeight + nextAttemptDelta
+		log.Debugf("Rescheduling input %v after %v attempts at height %v (delta %v)",
+			input.PreviousOutPoint, pi.publishAttempts,
+			pi.minPublishHeight, nextAttemptDelta,
+		)
+
+		if pi.publishAttempts >= s.cfg.MaxSweepAttempts {
+			// Signal result channels sweep result.
+			s.signalAndRemove(&input.PreviousOutPoint, Result{
+				Err: fmt.Errorf(
+					"sweep failed after %v attempts",
+					pi.publishAttempts,
+				),
+			},
```

Comment:
Can remove this extra line below by having the close bracket and the close paren be on the same line.